NetUtils.py

```diff
@@ -11,12 +11,87 @@
 from Utils import ByValue, Version
 
 
-class HintStatus(ByValue, enum.IntEnum):
-    HINT_UNSPECIFIED = 0
-    HINT_NO_PRIORITY = 10
-    HINT_AVOID = 20
-    HINT_PRIORITY = 30
-    HINT_FOUND = 40
+class HintStatus(ByValue, enum.IntFlag):
+    # Lower 5 bits: Priorities as integers
+    HINT_PRIORITY_UNSPECIFIED = 0  # For readable code
+    HINT_PRIORITY_NO_PRIORITY = 1
+    HINT_PRIORITY_AVOID = 2
+    HINT_PRIORITY_PRIORITY = 3
+    PRIORITY_MASK = 0b11111
+
+    # Bits 6+: Technical status
+    HINT_FOUND = 0b10000000
+
+    # For "compatibility", only used in CommonClient, thus should not affect memory for MultiServer
+    HINT_UNFOUND_LEGACY = OLD_HINT_FORMAT = 0b1000000000000000
```

Comment:
What would you think of using bit shifts here? I think it is easier to understand that the bit 15 is set with something like
```suggestion
    HINT_UNFOUND_LEGACY = OLD_HINT_FORMAT = 1 << 15
```


(same comment for `HINT_FOUND`)